### PR TITLE
Fix baseline identifier munging issue

### PR DESCRIPTION
**Pull request by bradcray**
This is a roundabout (but ultimately positive!) fix for a baseline failure in one of my identifier munging tests.  Specifically, my .prediff script looked for instances of foo_chpl (because the test was written such that there weren't intended to be any), but in --baseline mode, it found one because the compiler introduced a getter named "_bar_foo" and then the "munge all identifiers" pass would make this into "_bar_foo_chpl" resulting in a match against foo_chpl.  (This getter is inlined in the non-baseline case, which is why we didn't see it there).

The fix was the following:

1) Make this getter start with "chpl_" rather than simply "_" for goodness' sake...  We're trying to get away from using "_" as though it's a safe identifier since C doesn't guarantee that.

2) Make the munging pass skip over symbols that start with "chpl_" on the argument that they're already plenty unique and that the compiler/Chapel implementers introduced them.

This isn't a particularly satisfying fix for the baseline regression itself, but both changes independently of the regression feel positive and more like the world we want to be living in, so I'm not at all ashamed.

### CODE REVIEW

**Comment by bradcray** [comment]
@ronawho : Can you review this as an extension of our discussion and as my ongoing "munge identifiers" reviewer?

File: compiler/passes/codegen.cpp [hunk 1]
```diff
@@ -532,6 +532,14 @@ static void codegen_header_compilation_config() {
 
 static void protectNameFromC(Symbol* sym) {
   //
+  // Symbols that start with 'chpl_' are presumably compiler-generated
+  // and sufficiently unique not to require further munging.
```
Comment (ronawho) [inline compiler/passes/codegen.cpp]: I can't think of clean wording, but I think it should say compiler-generated or a symbol used by the internal modules

Comment (ronawho) [inline compiler/passes/codegen.cpp]: i.e. chpl_build_bounded_range is not compiler-generated

**Comment by ronawho** [comment]
Looks good other than that comment

**Comment by ronawho** [comment]
Oh wait... are we causing issues for ourselves here? Could prefixing with chpl_ cause us to change to a name that will conflict with an existing one. 

For example, if we already have a function in the modules called chpl_Foo_x, won't a compiler generated Foo_x that we then prefix cause issues?

**Comment by bradcray** [comment]
Here's my new proposed comment:

  //                                                                            
  // Symbols that start with 'chpl_' were presumably named by the               
  // implementation (compiler, internal modules, runtime) and                   
  // sufficiently unique to not require further munging.                        
  //                                                                            

I think you're right that there can be conflicts like this.  I think the solution is to use variations on "chpl_" for compiler-generated names.  E.g., Maybe I should call this "chpl_get[ter]" (and then, by convention, no internal modules/symbols should be named chpl_get[ter]).  I'm not sure that this problem is unique to this change, though -- i.e., I think other places in the compiler do similar things and run into similar issues.  I can make that change if you like.

I think in any case, the old "_Foo_x" (underscore only) is indefensible.

**Comment by ronawho** [comment]
I like the new comment. 

I also like the variations for compiler-generated names. I don't have a preference for if that change goes in now or not, I think this is a nice improvement even without that.

**Comment by bradcray** [comment]
This is a roundabout (but ultimately positive!) fix for a baseline failure in one of my identifier munging tests. Specifically, my .prediff script looked for instances of foo_chpl (because the test was written such that there weren't intended to be any), but in --baseline mode, it found one because the compiler introduced a getter named "_bar_foo" and then the "munge all identifiers" pass would make this into "_bar_foo_chpl" resulting in a match against foo_chpl. (This getter is inlined in the non-baseline case, which is why we didn't see it there).

The fix was the following:

1) Make this getter start with "chpl_get_" rather than simply "\_" for goodness' sake... We're trying to get away from using "\_" as though it's a safe identifier since C doesn't guarantee that.  The choice of "chpl_get_" suggests that this prefix shouldn't be used by other manually- or automatically-named symbols.

2) Make the munging pass skip over symbols that start with "chpl\_" on the argument that they're already plenty unique and that the compiler/Chapel implementers introduced them.

This isn't a particularly satisfying fix for the baseline regression itself, but both changes independently of the regression feel positive and more like the world we want to be living in, so I'm not at all ashamed.

**Pull request merged**